<?xml version="1.0"?>
<launch>
  <!-- Load parameters from configuration files -->
  <rosparam file="$(find bot)/config/controllers.yaml" command="load"/>
  <rosparam file="$(find bot)/config/joint_limits.yaml" command="load"/>

  <!-- Argument for the robot model file -->
  <arg name="model" default="$(find bot)/urdf/mobile_robot.urdf.xacro"/>
  
  <!-- Load the robot description parameter -->
  <param name="robot_description" command="/opt/ros/noetic/bin/xacro $(arg model)"/>

  <!-- Start the robot hardware interface node -->
  <node name="mobile_robot_hardware_interface" pkg="mobile_robot_navigation" type="robot_hardware_interface_node" output="screen">
    <remap from="/mobile_robot/mobile_base_controller/cmd_vel" to="/cmd_vel"/>
  </node>

  <!-- Start the robot state publisher -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>
  
  <!-- Start RViz with a specific configuration -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find bot)/config/rviz_config.rviz"/>
  
  <!-- Start the controller spawner -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"
        args="
          /mobile_robot/joints_update
          /mobile_robot/mobile_base_controller"/>
  
  <!-- Start the motor control node with USB connection -->
  <node name="motor_control" pkg="motor_control" type="motor_control.py" output="screen">
    <param name="port" type="string" value="/dev/ttyUSB0"/>
    <param name="baud" type="int" value="57600"/>
  </node>
</launch>
